asciidoc/courses/drivers-java/modules/1-driver/lessons/4-object-mapping/lesson.adoc

```diff
@@ -0,0 +1,139 @@
+= Mapping Results to Java Objects
+:type: lesson 
+:slides: true
+:minutes: 15
+:order: 5
+
+[.slide.col-2]
+== Introduction
+
+[.col]
+As of the Neo4j Java driver 5.28.5, the driver includes an link:https://github.com/neo4j/neo4j-java-driver/discussions/1640[object mapping feature] that allows you to map query results directly to Java objects. This feature simplifies the process of working with Neo4j data in Java applications by eliminating the need for manual mapping with raw data types.
+
+[.col]
+====
+[NOTE]
+.Object Graph Mapping (OGM)
+=====
+The driver's new object mapping is not a full-fledged OGM solution. For a more comprehensive option, check out the https://neo4j.com/docs/ogm-manual/current/[Neo4j OGM library^].
+=====
+====
+
+[.slide]
+== Domain Model
+
+For a person domain class, create a new file in `src/main/java/com/neo4j/app` and name it `Person.java`. This example will use Java records (rather than class), but classes works similarly.
+
+[source,Java]
+----
+public record Person(String id, 
+                     String name) {
+}
+----
+
+[.slide.col-2]
+== Querying and Mapping Results
+
+[.col]
+====
+Next, open the `src/main/java/com/neo4j/app/App.java` file and add a method to query for a person by name and return an entity mapped to your `Person` record:
+
+[source,Java]
+----
+final String personCypher = """
+    MATCH (person:Person {name: $name})
+    RETURN person
+    """;
+final String name = "Tom Hanks";
+
```

Comment:
I think this will need to be split into 2 slides. As it doesnt fit on 1.

Reply:
Does this look better?

Reply:
It does :)